.circleci/config.yml

```diff
@@ -0,0 +1,164 @@
+version: 2
+
+jobs:
+  checkout_code:
+    docker:
+      - image: jfullaondo/datadog-agent-builder-deb:0.0.2
+    working_directory: /go/src/github.com/DataDog/datadog-agent
+    steps:
+      - checkout
+      - run:
+          name: save SHA to a file
+          command: echo $CIRCLE_SHA1 > .circle-sha
+      - save_cache:
+          key: v1-repo-{{ checksum ".circle-sha" }}
```

Comment:
Is it worth it? I don't see that many cases where you would re-run the CI pipeline on the exact same codebase.

> keep in mind that cache saving is not a free operation, it will take some time to upload the cache to CircleCI storage